iirfilt_crcf_execute: how to change calculation method from df2 to df1

After browsing the source code, I found `iirfilt_crcf_execute` uses df2 method by default when filter format is sos. However, I would like to use df1 method in my case.

I came up with creating an `iirfiltsos` object instead of `iirfilt` but compiler complains about no such `iirfiltsos` object:
```
$ gcc iirfiltsos.c -lliquid -lm
iirfiltsos.c: In function ‘main’:
iirfiltsos.c:31:5: error: unknown type name ‘iirfiltsos_crcf’; did you mean ‘iirfilt_crcf’?
   31 |     iirfiltsos_crcf q = iirfiltsos_crcf_create(b, a);
      |     ^~~~~~~~~~~~~~~
      |     iirfilt_crcf
iirfiltsos.c:31:25: warning: implicit declaration of function ‘iirfiltsos_crcf_create’; did you mean ‘iirfilt_crcf_create’? [-Wimplicit-function-declaration]
   31 |     iirfiltsos_crcf q = iirfiltsos_crcf_create(b, a);
      |                         ^~~~~~~~~~~~~~~~~~~~~~
      |                         iirfilt_crcf_create
iirfiltsos.c:34:5: warning: implicit declaration of function ‘iirfiltsos_crcf_print’; did you mean ‘iirfilt_crcf_print’? [-Wimplicit-function-declaration]
   34 |     iirfiltsos_crcf_print(q);
      |     ^~~~~~~~~~~~~~~~~~~~~
      |     iirfilt_crcf_print
iirfiltsos.c:45:9: warning: implicit declaration of function ‘iirfiltsos_crcf_execute_df1’; did you mean ‘iirfilt_crcf_execute’? [-Wimplicit-function-declaration]
   45 |         iirfiltsos_crcf_execute_df1(q, x[i], &y[i]);
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
      |         iirfilt_crcf_execute
iirfiltsos.c:48:5: warning: implicit declaration of function ‘iirfiltsos_crcf_destroy’; did you mean ‘iirfilt_crcf_destroy’? [-Wimplicit-function-declaration]
   48 |     iirfiltsos_crcf_destroy(q);
      |     ^~~~~~~~~~~~~~~~~~~~~~~
      |     iirfilt_crcf_destroy
```
(The code snippet is here: https://github.com/Cuda-Chen/liquid-dsp-practice/blob/master/iirfiltsos/iirfiltsos.c)

So I want to know if there are some methods to use df1 method by either creating an `iirfiltsos` object correctly or choosing df1 method when calling `iirfilt_crcf_execute`?

--- a/include/liquid.h
+++ b/include/liquid.h
@@ -3053,6 +3053,83 @@ LIQUID_IIRFILT_DEFINE_API(LIQUID_IIRFILT_MANGLE_CCCF,
                           liquid_float_complex,
                           liquid_float_complex)
 
+//
+// iirfiltsos : infinite impulse respone filter (second-order sections)
+//
+#define LIQUID_IIRFILTSOS_MANGLE_RRRF(name)  LIQUID_CONCAT(iirfiltsos_rrrf,name)
+#define LIQUID_IIRFILTSOS_MANGLE_CRCF(name)  LIQUID_CONCAT(iirfiltsos_crcf,name)
+#define LIQUID_IIRFILTSOS_MANGLE_CCCF(name)  LIQUID_CONCAT(iirfiltsos_cccf,name)
+
+#define LIQUID_IIRFILTSOS_DEFINE_INTERNAL_API(IIRFILTSOS,TO,TC,TI)  \
+typedef struct IIRFILTSOS(_s) * IIRFILTSOS();                   \
+                                                                \
+/* create 2nd-order infinite impulse reponse filter         */  \
+/*  _b      : feed-forward coefficients [size: _3 x 1]      */  \
+/*  _a      : feed-back coefficients    [size: _3 x 1]      */  \
+IIRFILTSOS() IIRFILTSOS(_create)(TC * _b,                       \
+                                 TC * _a);                      \
+                                                                \
+/* explicitly set 2nd-order IIR filter coefficients         */  \
+/*  _q      : iirfiltsos object                             */  \
+/*  _b      : feed-forward coefficients [size: _3 x 1]      */  \
+/*  _a      : feed-back coefficients    [size: _3 x 1]      */  \
+void IIRFILTSOS(_set_coefficients)(IIRFILTSOS() _q,             \
+                                   TC *         _b,             \
+                                   TC *         _a);            \
+                                                                \
+/* destroy iirfiltsos object, freeing all internal memory   */  \
+void IIRFILTSOS(_destroy)(IIRFILTSOS() _q);                     \
+                                                                \
+/* print iirfiltsos object properties to stdout             */  \
+void IIRFILTSOS(_print)(IIRFILTSOS() _q);                       \
+                                                                \
+/* clear/reset iirfiltsos object internals                  */  \
+void IIRFILTSOS(_reset)(IIRFILTSOS() _q);                       \
+                                                                \
+/* compute filter output                                    */  \
+/*  _q      : iirfiltsos object                             */  \
+/*  _x      : input sample                                  */  \
+/*  _y      : output sample pointer                         */  \
+void IIRFILTSOS(_execute)(IIRFILTSOS() _q,                      \
+                          TI           _x,                      \
+                          TO *         _y);                     \
+                                                                \
+/* compute filter output, direct-form I method              */  \
+/*  _q      : iirfiltsos object                             */  \
+/*  _x      : input sample                                  */  \
+/*  _y      : output sample pointer                         */  \
+void IIRFILTSOS(_execute_df1)(IIRFILTSOS() _q,                  \
+                              TI           _x,                  \
+                              TO *         _y);                 \
+                                                                \
+/* compute filter output, direct-form II method             */  \
+/*  _q      : iirfiltsos object                             */  \
+/*  _x      : input sample                                  */  \
+/*  _y      : output sample pointer                         */  \
+void IIRFILTSOS(_execute_df2)(IIRFILTSOS() _q,                  \
+                              TI           _x,                  \
+                              TO *         _y);                 \
+                                                                \
+/* compute and return group delay of filter object          */  \
+/*  _q      : filter object                                 */  \
+/*  _fc     : frequency to evaluate                         */  \
+float IIRFILTSOS(_groupdelay)(IIRFILTSOS() _q,                  \
+                              float        _fc);                \
+
+LIQUID_IIRFILTSOS_DEFINE_INTERNAL_API(LIQUID_IIRFILTSOS_MANGLE_RRRF,
+                                      float,
+                                      float,
+                                      float)
+
+LIQUID_IIRFILTSOS_DEFINE_INTERNAL_API(LIQUID_IIRFILTSOS_MANGLE_CRCF,
+                                      liquid_float_complex,
+                                      float,
+                                      liquid_float_complex)
+
+LIQUID_IIRFILTSOS_DEFINE_INTERNAL_API(LIQUID_IIRFILTSOS_MANGLE_CCCF,
+                                      liquid_float_complex,
+                                      liquid_float_complex,
+                                      liquid_float_complex)
 
 //
 // FIR Polyphase filter bank

--- a/include/liquid.h
+++ b/include/liquid.h
@@ -3060,7 +3060,7 @@ LIQUID_IIRFILT_DEFINE_API(LIQUID_IIRFILT_MANGLE_CCCF,
 #define LIQUID_IIRFILTSOS_MANGLE_CRCF(name)  LIQUID_CONCAT(iirfiltsos_crcf,name)
 #define LIQUID_IIRFILTSOS_MANGLE_CCCF(name)  LIQUID_CONCAT(iirfiltsos_cccf,name)
 
-#define LIQUID_IIRFILTSOS_DEFINE_INTERNAL_API(IIRFILTSOS,TO,TC,TI)  \
+#define LIQUID_IIRFILTSOS_DEFINE_API(IIRFILTSOS,TO,TC,TI)  \
 typedef struct IIRFILTSOS(_s) * IIRFILTSOS();                   \
                                                                 \
 /* create 2nd-order infinite impulse reponse filter         */  \
@@ -3116,17 +3116,17 @@ void IIRFILTSOS(_execute_df2)(IIRFILTSOS() _q,                  \
 float IIRFILTSOS(_groupdelay)(IIRFILTSOS() _q,                  \
                               float        _fc);                \
 
-LIQUID_IIRFILTSOS_DEFINE_INTERNAL_API(LIQUID_IIRFILTSOS_MANGLE_RRRF,
+LIQUID_IIRFILTSOS_DEFINE_API(LIQUID_IIRFILTSOS_MANGLE_RRRF,
                                       float,
                                       float,
                                       float)
 
-LIQUID_IIRFILTSOS_DEFINE_INTERNAL_API(LIQUID_IIRFILTSOS_MANGLE_CRCF,
+LIQUID_IIRFILTSOS_DEFINE_API(LIQUID_IIRFILTSOS_MANGLE_CRCF,
                                       liquid_float_complex,
                                       float,
                                       liquid_float_complex)
 
-LIQUID_IIRFILTSOS_DEFINE_INTERNAL_API(LIQUID_IIRFILTSOS_MANGLE_CCCF,
+LIQUID_IIRFILTSOS_DEFINE_API(LIQUID_IIRFILTSOS_MANGLE_CCCF,
                                       liquid_float_complex,
                                       liquid_float_complex,
                                       liquid_float_complex)

--- a/include/liquid.internal.h
+++ b/include/liquid.internal.h
@@ -815,68 +815,6 @@ LIQUID_FIRFARROW_DEFINE_INTERNAL_API(LIQUID_FIRFARROW_MANGLE_CRCF,
 
 
 
-// 
-// iirfiltsos : infinite impulse respone filter (second-order sections)
-//
-#define LIQUID_IIRFILTSOS_MANGLE_RRRF(name)  LIQUID_CONCAT(iirfiltsos_rrrf,name)
-#define LIQUID_IIRFILTSOS_MANGLE_CRCF(name)  LIQUID_CONCAT(iirfiltsos_crcf,name)
-#define LIQUID_IIRFILTSOS_MANGLE_CCCF(name)  LIQUID_CONCAT(iirfiltsos_cccf,name)
-
-#define LIQUID_IIRFILTSOS_DEFINE_INTERNAL_API(IIRFILTSOS,TO,TC,TI)  \
-typedef struct IIRFILTSOS(_s) * IIRFILTSOS();                   \
-                                                                \
-/* create 2nd-order infinite impulse reponse filter         */  \
-/*  _b      : feed-forward coefficients [size: _3 x 1]      */  \
-/*  _a      : feed-back coefficients    [size: _3 x 1]      */  \
-IIRFILTSOS() IIRFILTSOS(_create)(TC * _b,                       \
-                                 TC * _a);                      \
-                                                                \
-/* explicitly set 2nd-order IIR filter coefficients         */  \
-/*  _q      : iirfiltsos object                             */  \
-/*  _b      : feed-forward coefficients [size: _3 x 1]      */  \
-/*  _a      : feed-back coefficients    [size: _3 x 1]      */  \
-void IIRFILTSOS(_set_coefficients)(IIRFILTSOS() _q,             \
-                                   TC *         _b,             \
-                                   TC *         _a);            \
-                                                                \
-/* destroy iirfiltsos object, freeing all internal memory   */  \
-void IIRFILTSOS(_destroy)(IIRFILTSOS() _q);                     \
-                                                                \
-/* print iirfiltsos object properties to stdout             */  \
-void IIRFILTSOS(_print)(IIRFILTSOS() _q);                       \
-                                                                \
-/* clear/reset iirfiltsos object internals                  */  \
-void IIRFILTSOS(_reset)(IIRFILTSOS() _q);                       \
-                                                                \
-/* compute filter output                                    */  \
-/*  _q      : iirfiltsos object                             */  \
-/*  _x      : input sample                                  */  \
-/*  _y      : output sample pointer                         */  \
-void IIRFILTSOS(_execute)(IIRFILTSOS() _q,                      \
-                          TI           _x,                      \
-                          TO *         _y);                     \
-                                                                \
-/* compute filter output, direct-form I method              */  \
-/*  _q      : iirfiltsos object                             */  \
-/*  _x      : input sample                                  */  \
-/*  _y      : output sample pointer                         */  \
-void IIRFILTSOS(_execute_df1)(IIRFILTSOS() _q,                  \
-                              TI           _x,                  \
-                              TO *         _y);                 \
-                                                                \
-/* compute filter output, direct-form II method             */  \
-/*  _q      : iirfiltsos object                             */  \
-/*  _x      : input sample                                  */  \
-/*  _y      : output sample pointer                         */  \
-void IIRFILTSOS(_execute_df2)(IIRFILTSOS() _q,                  \
-                              TI           _x,                  \
-                              TO *         _y);                 \
-                                                                \
-/* compute and return group delay of filter object          */  \
-/*  _q      : filter object                                 */  \
-/*  _fc     : frequency to evaluate                         */  \
-float IIRFILTSOS(_groupdelay)(IIRFILTSOS() _q,                  \
-                              float        _fc);                \
 
 LIQUID_IIRFILTSOS_DEFINE_INTERNAL_API(LIQUID_IIRFILTSOS_MANGLE_RRRF,
                                       float,

--- a/include/liquid.internal.h
+++ b/include/liquid.internal.h
@@ -814,24 +814,6 @@ LIQUID_FIRFARROW_DEFINE_INTERNAL_API(LIQUID_FIRFARROW_MANGLE_CRCF,
                                      liquid_float_complex)
 
 
-
-
-LIQUID_IIRFILTSOS_DEFINE_INTERNAL_API(LIQUID_IIRFILTSOS_MANGLE_RRRF,
-                                      float,
-                                      float,
-                                      float)
-
-LIQUID_IIRFILTSOS_DEFINE_INTERNAL_API(LIQUID_IIRFILTSOS_MANGLE_CRCF,
-                                      liquid_float_complex,
-                                      float,
-                                      liquid_float_complex)
-
-LIQUID_IIRFILTSOS_DEFINE_INTERNAL_API(LIQUID_IIRFILTSOS_MANGLE_CCCF,
-                                      liquid_float_complex,
-                                      liquid_float_complex,
-                                      liquid_float_complex)
-
-
 // firdes : finite impulse response filter design
 
 // Find approximate bandwidth adjustment factor rho based on

